src/shared/useAccessibilityAutoFocus.ts

```diff
@@ -0,0 +1,47 @@
+import {useLayoutEffect, useState} from 'react';
+import {AccessibilityInfo, findNodeHandle} from 'react-native';
+
+import {createCancellableCallbackPromise} from './cancellablePromise';
+
+/**
+ * Look like there is timing issue with AccessibilityInfo.setAccessibilityFocus
+ * Ref https://github.com/react-native-community/discussions-and-proposals/issues/118
+ */
+const AUTO_FOCUS_DELAY = 200;
+
+/**
+ * Remember to add `accessible={true}` to target component
+ */
+export const useAccessibilityAutoFocus = (isActive = true) => {
+  const [autoFocusRef, setAutoFocusRef] = useState<any>();
+  useLayoutEffect(() => {
+    if (!autoFocusRef || !isActive) {
+      return;
+    }
+    const {callable, cancelable} = createCancellableCallbackPromise(
+      () => AccessibilityInfo.isScreenReaderEnabled().then(delay(AUTO_FOCUS_DELAY)),
```

Comment:
Second, we only do the timeout and focus if screenReader is enabled.

Reply:
Adding the timeout 200ms seems to be more stable. `createCancellableCallbackPromise` will make sure if component is unmounted before timeout is finished, it won't leak.